tests/spark_autologging/test_pyspark_ml_autologging.py

```diff
@@ -0,0 +1,222 @@
+        "{} in log_model_allowlist don't exist".format(classes_does_not_exist)
+
+
+def test_autolog_preserves_original_function_attributes():
+    from pyspark.ml import Estimator
+
+    def get_func_attrs(f):
+        attrs = {}
+        for attr_name in ["__doc__", "__name__"]:
+            if hasattr(f, attr_name):
+                attrs[attr_name] = getattr(f, attr_name)
+
+        attrs["__signature__"] = inspect.signature(f)
+        return attrs
+
+    before = get_func_attrs(getattr(Estimator, 'fit'))
+    mlflow.pyspark.ml.autolog()
+    after = get_func_attrs(getattr(Estimator, 'fit'))
+
+    for b, a in zip(before, after):
+        assert b == a
+
+
+def test_autolog_does_not_terminate_active_run(dataset_binary):
+    mlflow.pyspark.ml.autolog()
+    mlflow.start_run()
+    lr = LinearRegression()
+    lr.fit(dataset_binary)
+    assert mlflow.active_run() is not None
+    mlflow.end_run()
+
+
+def test_meta_estimator_fit_performs_logging_only_once(dataset_binary):
+    from pyspark.ml.classification import LogisticRegression, OneVsRest
+    mlflow.pyspark.ml.autolog()
+    with mock.patch("mlflow.log_params") as mock_log_params, \
+            mock.patch("mlflow.set_tags") as mock_set_tags, \
+            mock.patch("mlflow.spark.log_model") as mock_log_model:
+        with mlflow.start_run() as run:
+            lor = LogisticRegression()
+            ova = OneVsRest(classifier=lor)
+            ova.fit(dataset_binary)
+
+        mock_log_params.assert_called_once()
+        mock_set_tags.assert_called_once()
+        mock_log_model.assert_called_once()
+
+        query = "tags.{} = '{}'".format(MLFLOW_PARENT_RUN_ID, run.info.run_id)
+        assert len(mlflow.search_runs([run.info.experiment_id])) == 1
+        assert len(mlflow.search_runs([run.info.experiment_id], query)) == 0
+
+
+def test_fit_with_params(dataset_binary):
+    mlflow.pyspark.ml.autolog()
+    lr = LinearRegression()
+    extra_params = {lr.maxIter: 3, lr.standardization: False}
+    extra_params2 = {lr.maxIter: 4, lr.standardization: True}
+    # Test calling fit with extra params
+    with mlflow.start_run() as run:
+        lr.fit(dataset_binary, params=extra_params)
+    run_id = run.info.run_id
+    run_data = get_run_data(run_id)
+    assert run_data.params == truncate_param_dict(stringify_dict_values(
+        _get_instance_param_map(lr.copy(extra_params))))
+
+    # Test calling fit with a list/tuple of paramMap
+    with mock.patch("mlflow.log_params") as mock_log_params, \
+            mock.patch("mlflow.set_tags") as mock_set_tags, \
+            mock.patch("mlflow.spark.log_model") as mock_log_model:
+
+        with mlflow.start_run() as run:
+            lr.fit(dataset_binary, params=[extra_params, extra_params2])
+            lr.fit(dataset_binary, params=(extra_params, extra_params2))
+
+        mock_log_params.assert_not_called()
+        mock_set_tags.assert_not_called()
+        mock_log_model.assert_not_called()
```

Comment:
Can we move this case in a separate test function and add assertion for the warning?